src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryReader.cs

```diff
@@ -9,15 +10,15 @@ namespace System.Xml
     public interface IXmlBinaryReaderInitializer
     {
         void SetInput(byte[] buffer, int offset, int count,
-                            IXmlDictionary dictionary,
+                            IXmlDictionary? dictionary,
                             XmlDictionaryReaderQuotas quotas,
-                            XmlBinaryReaderSession session,
-                            OnXmlDictionaryReaderClose onClose);
+                            XmlBinaryReaderSession? session,
+                            OnXmlDictionaryReaderClose? onClose);
```

Comment:
Seems like if `dictionary` or `session` are null, we throw an exception later in the code, is it OK that these parameters are nullable?
https://github.com/dotnet/runtime/blob/9b79ee6db16b27d10a49c6f73b54cf13c38bdcaa/src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs#L1146-L1165

Reply:
These parameters have to be nullable because we have these public overloads that explicitly pass `null` to these two parameters:

https://github.com/dotnet/runtime/blob/d5d046d03589dedc4288515ddde90b3e804c3dc2/src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlDictionaryReader.cs#L44-L52

Reply:
Ah, right.